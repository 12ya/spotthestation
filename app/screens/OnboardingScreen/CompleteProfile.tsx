import { useNavigation } from "@react-navigation/native"
import React, { useCallback, useState } from "react"
import { TextStyle, View, ViewStyle } from "react-native"
import { Screen, Text } from "../../components"
import { typography } from "../../theme"
import { colors } from "../../theme/colors"
import { useSafeAreaInsetsStyle } from "../../utils/useSafeAreaInsetsStyle"
import { IconLinkButton } from "./components/IconLinkButton"
import { LocationType, SignupLocation } from "./SignupLocation"
import { SignupNotificationSettings } from "./SignupNotificationSettings"
import * as storage from "../../utils/storage"

export function CompleteProfile() {
  const navigation = useNavigation()

  const $topInset = useSafeAreaInsetsStyle(["top", "bottom"], "padding")

  const [step, setStep] = useState(1)
  const [notifications, setNotifications] = useState(false)
  const [location, setLocation] = useState({
    title: "",
    subtitle: "",
    location: {
      lat: null,
      lng: null
    }
  })

  const handleBack = useCallback(() => {
    if (step === 1) navigation.goBack()
    else setStep(step - 1)
  }, [step])

  const handleNext = useCallback(() => setStep(step + 1), [step])

  const handleDone = () => navigation.navigate("Main" as never)

  const handleLocationChange = async (location: LocationType) => {
    setLocation(location)
    await storage.save('currentLocation', location)
  }

  const renderBody = useCallback(() => {
    switch(step) {
      case 2: return <SignupLocation value={location} onValueChange={handleLocationChange} onAction={handleDone} />
      default: return <SignupNotificationSettings value={notifications} onValueChange={setNotifications} onAction={handleNext} />
    }
  }, [step, notifications, location])

  return (
    <Screen preset="scroll" contentContainerStyle={$container} style={[$topInset, {backgroundColor: colors.palette.neutral900}]} statusBarStyle="light">
      <IconLinkButton 
        icon="back"
        accessible
        accessibilityLabel="Back button"
        accessibilityHint="Navigates to the previous screen"
        onPress={handleBack} 
        buttonStyle={$back} 
      />
      <View style={$contentContainer}>
        <Text
          accessible
          accessibilityLabel="Step counter"
          accessibilityHint="Display current step"
          accessibilityRole="text"
          text={`${step} of 2`} 
          style={$step}
        />
        {renderBody()}
      </View>
    </Screen>
  )
}

const $container: ViewStyle = {
  flex: 1,
  backgroundColor: colors.palette.neutral900,
  paddingTop: 24
}

const $contentContainer: ViewStyle = {
  flex: 1,
  paddingHorizontal: 36,
  paddingBottom: 52
}

const $back: ViewStyle = {
  marginLeft: 18
}

const $step: TextStyle = {
  color: colors.palette.neutral450,
  fontSize: 18,
  fontFamily: typography.primary.normal,
  lineHeight: 22,
  paddingBottom: 18,
  paddingTop: 36
}
