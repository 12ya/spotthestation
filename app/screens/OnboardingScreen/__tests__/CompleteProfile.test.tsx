/* eslint-disable react/display-name */
/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
import { NavigationContainer } from "@react-navigation/native"
import React from "react"
import { CompleteProfile } from "../CompleteProfile"
import { render } from "@testing-library/react-native"

it("renders correctly", () => {
  const tree = render(
    <NavigationContainer>
      <CompleteProfile />
    </NavigationContainer>,
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
