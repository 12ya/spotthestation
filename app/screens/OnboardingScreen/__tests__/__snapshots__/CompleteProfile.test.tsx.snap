// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <View
    accessibilityHint="Navigates to the previous screen"
    accessibilityLabel="Back button"
    accessibilityRole="imagebutton"
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#1E2022",
          "borderRadius": 143,
          "display": "flex",
          "height": 60,
          "justifyContent": "center",
          "overflow": "hidden",
          "position": "relative",
          "width": 60,
        },
        Object {
          "marginLeft": 26,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#1E2022",
              "borderRadius": 143,
              "display": "flex",
              "height": 60,
              "justifyContent": "center",
              "overflow": "hidden",
              "position": "relative",
              "width": 60,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      <ViewManagerAdapter_ExpoBlurView
        proxiedProperties={
          Object {
            "intensity": 0,
            "tint": "default",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
      <View>
        <RCTImageView
          internal_analyticTag={null}
          resizeMode="contain"
          source={
            Array [
              Object {
                "height": undefined,
                "uri": undefined,
                "width": undefined,
              },
            ]
          }
          style={
            Object {
              "height": 29,
              "overflow": "hidden",
              "resizeMode": "contain",
              "tintColor": "#FFFFFF",
              "width": 29,
            }
          }
          tintColor="#FFFFFF"
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "paddingBottom": 75,
        "paddingHorizontal": 52,
      }
    }
  >
    <Text
      accessibilityHint="Display current step"
      accessibilityLabel="Step counter"
      accessibilityRole="text"
      accessible={true}
      allowFontScaling={false}
      style={
        Array [
          Object {},
          Array [
            Object {
              "fontSize": 23,
              "lineHeight": 35,
            },
            Object {
              "fontFamily": "Helvetica Neue",
            },
            Object {
              "color": "#191015",
            },
          ],
          undefined,
          undefined,
          Object {
            "color": "#ADADAE",
            "fontFamily": "Helvetica Neue",
            "fontSize": 26,
            "lineHeight": 32,
            "paddingBottom": 26,
            "paddingTop": 52,
          },
        ]
      }
    >
      1 of 2
    </Text>
    <Text
      accessibilityHint="detecting current location"
      accessibilityLabel="detecting"
      accessibilityRole="text"
      accessible={true}
      allowFontScaling={false}
      style={
        Array [
          Object {},
          Array [
            Object {
              "fontSize": 23,
              "lineHeight": 35,
            },
            Object {
              "fontFamily": "Helvetica Neue",
            },
            Object {
              "color": "#191015",
            },
          ],
          undefined,
          undefined,
          Object {
            "color": "#DEDDDE",
            "fontFamily": "Helvetica Neue",
            "fontSize": 52,
            "lineHeight": 60,
            "paddingBottom": 52,
          },
        ]
      }
    >
      onboarding.completeProfile.notification.title undefined
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 52,
        }
      }
    >
      <View
        style={
          Object {
            "marginRight": 26,
          }
        }
      >
        <RCTImageView
          internal_analyticTag={null}
          resizeMode="contain"
          source={
            Array [
              Object {
                "height": undefined,
                "uri": undefined,
                "width": undefined,
              },
            ]
          }
          style={
            Object {
              "height": 35,
              "overflow": "hidden",
              "resizeMode": "contain",
              "width": 35,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#1E2022",
            "borderBottomWidth": 2,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 26,
          }
        }
      >
        <View
          accessibilityHint="notification"
          accessibilityLabel="notification"
          accessibilityRole="text"
          accessible={true}
          style={
            Object {
              "width": "70%",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {},
                Array [
                  Object {
                    "fontSize": 23,
                    "lineHeight": 35,
                  },
                  Object {
                    "fontFamily": "Helvetica Neue",
                  },
                  Object {
                    "color": "#191015",
                  },
                ],
                undefined,
                undefined,
                Object {
                  "color": "#DEDDDE",
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 23,
                  "lineHeight": 30,
                },
              ]
            }
          >
            onboarding.completeProfile.notification.label undefined
          </Text>
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {},
                Array [
                  Object {
                    "fontSize": 23,
                    "lineHeight": 35,
                  },
                  Object {
                    "fontFamily": "Helvetica Neue",
                  },
                  Object {
                    "color": "#191015",
                  },
                ],
                undefined,
                undefined,
                Object {
                  "color": "#ADADAE",
                  "fontFamily": "HelveticaNeue-Light",
                  "fontSize": 23,
                  "lineHeight": 30,
                },
              ]
            }
          >
            onboarding.completeProfile.notification.tip undefined
          </Text>
        </View>
        <View
          accessibilityHint="toggle notifications"
          accessibilityLabel="switch button"
          accessibilityRole="switch"
          accessibilityState={
            Object {
              "checked": false,
              "disabled": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderWidth": 3,
                      "flexDirection": "row",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "height": 35,
                      "justifyContent": "space-between",
                      "overflow": "hidden",
                      "width": 35,
                    },
                    Object {
                      "borderRadius": 23,
                      "borderWidth": 0,
                      "height": 40,
                      "width": 66,
                    },
                  ],
                  Object {
                    "backgroundColor": "#353739",
                  },
                  undefined,
                ]
              }
            >
              <View
                animatedStyle={
                  Object {
                    "value": Object {
                      "opacity": 0,
                    },
                  }
                }
                collapsable={false}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#0B3D91",
                    "borderColor": "rgba(0, 0, 0, 0)",
                    "height": "100%",
                    "opacity": 0,
                    "overflow": "hidden",
                    "paddingEnd": 6,
                    "paddingStart": 6,
                    "position": "absolute",
                    "width": "100%",
                  }
                }
              />
              <View
                animatedStyle={
                  Object {
                    "value": Object {
                      "marginStart": 6,
                      "start": "0%",
                    },
                  }
                }
                collapsable={false}
                style={
                  Object {
                    "backgroundColor": "#F4F2F1",
                    "borderRadius": 17,
                    "height": 29,
                    "marginStart": 6,
                    "position": "absolute",
                    "start": "0%",
                    "width": 29,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityHint="navigate to location detection"
      accessibilityLabel="next button"
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 6,
              "flexDirection": "row",
              "justifyContent": "center",
              "minHeight": 80,
              "overflow": "hidden",
              "paddingHorizontal": 17,
              "paddingVertical": 17,
            },
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#B6ACA6",
              "borderWidth": 1,
            },
          ],
          Object {
            "backgroundColor": "#0B3D91",
            "borderRadius": 40,
            "borderWidth": 0,
            "height": 80,
            "width": "100%",
          },
          false,
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {},
            Array [
              Object {
                "fontSize": 23,
                "lineHeight": 35,
              },
              Object {
                "fontFamily": "Helvetica Neue",
              },
              Object {
                "color": "#191015",
              },
            ],
            undefined,
            undefined,
            Array [
              Object {
                "flexGrow": 0,
                "flexShrink": 1,
                "fontFamily": "HelveticaNeue-Medium",
                "fontSize": 23,
                "lineHeight": 29,
                "textAlign": "center",
                "zIndex": 2,
              },
              Object {
                "color": "#FFFFFF",
                "fontFamily": "HelveticaNeue-Medium",
                "fontSize": 26,
                "lineHeight": 30,
              },
              false,
            ],
          ]
        }
      >
        onboarding.completeProfile.notification.nextButton undefined
      </Text>
    </View>
  </View>
</div>
`;
